WordPress/wporg-news-2021
# Add an "All Posts" template to list full archive

See #71, #284. Switch homepage and front page to use static pages, so that we can have a separate "All Posts" page. This also simplifies some of the logic around template loading.

New testing instructions in https://github.com/WordPress/wporg-news-2021/pull/287#issuecomment-1031885548

<details>
<summary>Old description</summary>
This adds a new template which is set on a page called "All Posts". When used, it shows a list of all the posts on the site in a simple list, 10 posts per page. It uses the same design & class names as the existing paged archive, but this one also shows the first page of posts too.

To test:

- Create a new page called "All Posts", the slug should be `all-posts` for the links to work
- Set the page template to "All Posts", don't worry about content. Publish it.
- View the page, it should show you a list of the latest posts.
- Use the pagination, it should correctly nav through older posts
- The "All Posts" item in the categories nav should stay highlighted

Note: I didn't remove any of this style logic from the `/page/n` pages, since power users might still visit those, they will look the same.
</details>

<details>
<summary>Full page screenshot</summary>

![Screen Shot 2022-02-04 at 15 46 47](https://user-images.githubusercontent.com/541093/152603260-ad1cecfd-3428-46ec-b379-508dda0199e4.png)

</details>
